crates/astria-composer/src/executor/builder.rs

```diff
@@ -36,7 +38,7 @@ pub(crate) struct Builder {
 }
 
 impl Builder {
-    pub(crate) fn build(self) -> eyre::Result<(super::Executor, executor::Handle)> {
+    pub(crate) async fn build(self) -> eyre::Result<(super::Executor, executor::Handle)> {
```

Comment:
Please don't make the builder async. If one needs config-like input, then move this into the `run`/`run_until_stopped` function and run it with a retry.